src/cloudflare/internal/vectorize-api.ts

```diff
@@ -16,144 +16,164 @@ enum Operation {
   VECTOR_DELETE = 5,
 }
 
-class VectorizeIndexImpl implements VectorizeIndex {
+/*
+ * The Vectorize beta VectorizeIndex shares the same methods, so to keep things simple, they share one implementation.
```

Comment:
This is what we discussed @mikea and I this holds true, I think we're happy with this approach!